packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { store as blocksStore } from '@wordpress/blocks';
+import {
+	PanelBody,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../store';
+import { unlock } from '../lock-unlock';
+import InspectorControls from '../components/inspector-controls';
+
+export const BlockBindingsPanel = () => {
+	const { bindings } = useSelect( ( select ) => {
+		const { getSelectedBlockClientId, getBlock } =
+			select( blockEditorStore );
+		const _selectedBlockClientId = getSelectedBlockClientId();
+
+		return {
+			bindings: getBlock( _selectedBlockClientId ).attributes?.metadata
+				?.bindings,
+		};
+	}, [] );
+
+	const sources = useSelect( ( select ) =>
+		unlock( select( blocksStore ) ).getAllBlockBindingsSources()
+	);
```

Comment:
Is there a specific reason to have two different `useSelect` hooks? Can't we run this logic in the previous one and return both sources and bindings?

Reply:
They should be combined to improve perf. See more in https://developer.wordpress.org/news/2024/03/28/how-to-work-effectively-with-the-useselect-hook/.